apps/src/music/views/ValidatorProvider.tsx

```diff
@@ -0,0 +1,24 @@
+import {ProgressManagerContext} from '@cdo/apps/labs/progress/ProgressContainer';
+import {Validator} from '@cdo/apps/labs/progress/ProgressManager';
+import React, {useContext, useEffect} from 'react';
+
+/**
+ * Utility component for providing a {@link Validator} to the current {@link ProgressManager}.
+ * This is mostly to avoid using more complex React context syntax in class components
+ * (like MusicView).
+ */
+const ValidatorProvider: React.FunctionComponent<{validator: Validator}> = ({
```

Comment:
do you think this component will be Music Lab specific? If not, should we move it to `/labs`?

Reply:
Yeah, this can move to `/labs` - it's mostly useful for class components (which MusicView is), but might be nice to have for other labs as well.